HeterogeneousCore/SonicTriton/scripts/cmsTriton

```diff
@@ -315,14 +319,59 @@ wait_server(){
 	echo "server is ready!"
 }
 
+edit_model(){
+    cp -r $1 $TMPDIR/local_model_repo/
+    IFS='/' read -ra ADDR <<< "$1"
+    CONFIG=$TMPDIR/local_model_repo/${ADDR[-1]}/config.pbtxt
+
+    echo "" >> "$CONFIG"
+    echo "instance_group [" >> "$CONFIG"
+    echo "  {" >> "$CONFIG"
+    echo "    count: "$2 >> "$CONFIG"
+    echo "    kind: KIND_CPU" >> "$CONFIG"
+    echo "  }" >> "$CONFIG"
+    echo "]" >> "$CONFIG"
+    echo "" >> "$CONFIG"
+
+    if grep -Fq "onnx" "$CONFIG"
+    then
+	echo "parameters { key: \"intra_op_thread_count\" value: { string_value: \"1\" } }" >> "$CONFIG"
+	echo "parameters { key: \"inter_op_thread_count\" value: { string_value: \"1\" } }" >> "$CONFIG"
+    fi
+
+    if grep -Fq "tensorflow" "$CONFIG"
+    then
+	echo "parameters { key: \"TF_NUM_INTRA_THREADS\" value: { string_value: \"1\" } }" >> "$CONFIG"
+	echo "parameters { key: \"TF_NUM_INTER_THREADS\" value: { string_value: \"1\" } }" >> "$CONFIG"
+	echo "parameters { key: \"TF_USE_PER_SESSION_THREADS\" value: { string_value: \"1\" } }" >> "$CONFIG"
+    fi
+}
+
 list_models(){
 	# make list of model repositories
+        if [ "$INSTANCES" -gt 0 ]
+	then
+	    if [ -d "$TMPDIR/local_model_repo" ]
+	    then
+		#Want to start with a fresh copy of model files in case this directory already exists with local edits
+		rm -rf $TMPDIR/local_model_repo
+		mkdir $TMPDIR/local_model_repo
+	    else
+		mkdir $TMPDIR/local_model_repo
+	    fi
+	fi
 	for MODEL in ${MODELS[@]}; do
 		# check if file was provided rather than directory
 		if [ -f "$MODEL" ]; then
 			MODEL="$(dirname "$MODEL")"
 		fi
-		REPOS+=("$(dirname "$MODEL")")
+		if [ "$INSTANCES" -gt 0 ]
+		then
+		    edit_model $MODEL "$INSTANCES"
+		    REPOS=$TMPDIR/local_model_repo
```

Comment:
just assign to `REPOS` once outside of (after) the loop over models

Reply:
Done